### ISSUE TITLE
text command does not preserve order of inline element

### ISSUE THREAD

**Issue by Lakitna** [Lakitna/cypress-commands#58]
# Describe the bug

Given the following situation:

```html
<p>Lorum <b>ipsum</b> dolor sit amet.</p>
```
```js
cy.get('p').text({ depth: Infinity })
```

Will yield:

```js
"Lorum dolor sit amet. ipsum"
```

That is an unexpected order.

# Expected behavior

It yields

```js
"Lorum ipsum dolor sit amet."
```

# Solution

The following change to the `text()` command seems to solve the issue.

```js
/**
 * @param {JQuery} element
 * @param {number} depth
 * @return {string}
 */
function getTextOfElement(element, depth) {
    return element
        .contents()
        .filter((_, content) => {
            console.log(content)
            // Only keep the text nodes
            return content.nodeType === Node.TEXT_NODE
                || (content.nodeType === Node.ELEMENT_NODE && depth > 0);
        })
        .map((_, content) => {
            if (content.nodeType === Node.TEXT_NODE) {
                // Get the text from the text node
                return content.data.trim();
            }
            if (content.nodeType === Node.ELEMENT_NODE) {
                // Get the text from child element
                return getTextOfElement($(content), --depth)
            }
        })
        .toArray()
        .join(' ')
        .trim();
```

**Comment by Lakitna** [comment]
This is a breaking change

**Comment by Lakitna** [comment]
Already fixed for #46
